### PR TITLE
<FileUpload/> - support children as a function

### PR DESCRIPTION
**Pull request by bamai**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/FileUpload/FileUpload.js [hunk 1]
```diff
@@ -21,14 +21,7 @@ class FileUpload extends React.PureComponent {
...
       <label className={className} data-hook={dataHook}>
-        {React.cloneElement(children, {
-          onClick: event => {
-            if (typeof children.props.onClick === 'function')
-              children.props.onClick();
-            this.inputRef.current.click();
-            event.preventDefault();
-          },
-        })}
+        {children(() => this.inputRef.current.click())}
```
Comment (shlomitc) [inline src/FileUpload/FileUpload.js:24]: I think it's better to pass the render prop as an object 
`children({upload: () => this.inputRef.current.click()})`
This makes it easier to extend in the future if needed

File: src/FileUpload/FileUpload.js [hunk 2]
```diff
@@ -54,7 +47,7 @@ FileUpload.propTypes = {
   className: PropTypes.string,
 
   /** An element to be wrapped by the component */
```
Comment (shlomitc) [inline src/FileUpload/FileUpload.js:49]: Let's add here a comment explaining it is a render prop that gets the method in props

File: src/FileUpload/docs/examples.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 export const simple = `
 <FileUpload onChange={files => console.log(files)}>
-  <Button>Upload a File</Button>
+  {fileUpload => (<Button onClick={() => {fileUpload.upload()}}>Upload a File</Button>)}
```
Comment (EgoziE) [inline src/FileUpload/docs/examples.js:3]:
```suggestion
  {({ openFileUploadDialog })=> <Button onClick={openFileUploadDialog}>Upload a File</Button>}
```

File: src/FileUpload/docs/examples.js [hunk 4]
```diff
@@ -9,6 +9,6 @@ export const withMultipleAndAccept = `
   multiple
   accept=".jpeg,.gif,.png"
 >
-  <TextButton prefixIcon={<Icons.Image />}>Upload Images</TextButton>
+{fileUpload => (<TextButton prefixIcon={<Icons.Image />} onClick={() => {fileUpload.upload()}}>Upload Images</TextButton>)}
```
Comment (EgoziE) [inline src/FileUpload/docs/examples.js:12]:
```suggestion
  {({ openFileUploadDialog })=> (
    <TextButton prefixIcon={<Icons.Image />} onClick={openFileUploadDialog}>
      Upload Images
    </TextButton>
  )}
```

File: src/FileUpload/FileUpload.js [hunk 5]
```diff
@@ -21,14 +21,7 @@ class FileUpload extends React.PureComponent {
...
       <label className={className} data-hook={dataHook}>
-        {React.cloneElement(children, {
-          onClick: event => {
-            if (typeof children.props.onClick === 'function')
-              children.props.onClick();
-            this.inputRef.current.click();
-            event.preventDefault();
-          },
-        })}
+        {children({ upload: () => this.inputRef.current.click() })}
```
Comment (EgoziE) [inline src/FileUpload/FileUpload.js:24]: `upload` does not indicate what this function really does.
I suggest `openFileUploadDialog`:

```suggestion
        {children({ openFileUploadDialog: () => this.inputRef.current.click() })}
```

File: src/FileUpload/test/FileUpload.spec.js [hunk 6]
```diff
@@ -13,7 +13,15 @@ describe(FileUpload.displayName, () => {
...
-      <Button>upload file</Button>
+      {fileUpload => (
+        <Button
+          onClick={() => {
+            fileUpload.upload();
+          }}
+        >
+          upload file
+        </Button>
+      )}
```
Comment (EgoziE) [inline src/FileUpload/test/FileUpload.spec.js:24]:
```suggestion
      {({ openFileUploadDialog })=> (
        <Button
          onClick={openFileUploadDialog}
        >
          upload file
        </Button>
      )}
```

File: src/FileUpload/test/FileUpload.tsx [hunk 7]
```diff
@@ -8,9 +8,13 @@ import * as puppeteer from 'puppeteer';
 import Button from '../../Button';
 
 function fileUploadWithMandatoryProps() {
-  return <FileUpload onChange={() => {}} children={<Button />} />;
+  return (
+    <FileUpload
+      onChange={() => {}}
+      children={({ openFileUpload }) => <Button onClick={openFileUpload} />}
```
Comment (EgoziE) [inline src/FileUpload/test/FileUpload.tsx:14]:
```suggestion
      children={({ openFileUploadDialog }) => <Button onClick={openFileUploadDialog} />}
```

File: src/FileUpload/test/FileUpload.tsx [hunk 8]
```diff
@@ -20,7 +24,7 @@ function fileUploadWithAllProps() {
       accept="jpg"
       capture="user"
       multiple
-      children={<div />}
+      children={({ openFileUpload }) => <Button onClick={openFileUpload} />}
```
Comment (EgoziE) [inline src/FileUpload/test/FileUpload.tsx:27]:
```suggestion
      children={({ openFileUploadDialog }) => <Button onClick={openFileUploadDialog} />}
```

File: src/FileUpload/FileUpload.js (hunk 5 above)
Comment (shlomitc) [inline src/FileUpload/FileUpload.js:24]:
```suggestion
        {children({ openFileUpload: () => this.inputRef.current.click() })}
```

File: src/FileUpload/test/FileUpload.spec.js [hunk 9]
```diff
@@ -13,7 +13,15 @@ describe(FileUpload.displayName, () => {
 
   const fileUploadExample = props => (
     <FileUpload onChange={() => {}} {...props}>
-      <Button>upload file</Button>
+      {fileUpload => (
+        <Button
+          onClick={() => {
+            fileUpload.upload();
```
Comment (shlomitc) [inline src/FileUpload/test/FileUpload.spec.js:19]:
```suggestion
            fileUpload. openFileUpload();
```

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5796/

**Pull request merged**